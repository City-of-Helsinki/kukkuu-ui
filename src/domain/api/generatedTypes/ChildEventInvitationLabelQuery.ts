/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEventInvitationLabelQuery
// ====================================================

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventGroupNode {
  /**
   * The ID of the object.
   */
  id: string;
  canChildEnroll: boolean | null;
}

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventNode {
  /**
   * The ID of the object.
   */
  id: string;
  canChildEnroll: boolean | null;
}

export type ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node =

    | ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventGroupNode
    | ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node_EventNode;

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges {
  /**
   * The item at the end of the edge
   */
  node: ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges_node | null;
}

export interface ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups {
  /**
   * Contains the nodes in this connection.
   */
  edges: (ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups_edges | null)[];
}

export interface ChildEventInvitationLabelQuery_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * All upcoming events and event groups for the child's project.
   */
  upcomingEventsAndEventGroups: ChildEventInvitationLabelQuery_child_upcomingEventsAndEventGroups | null;
}

export interface ChildEventInvitationLabelQuery {
  child: ChildEventInvitationLabelQuery_child | null;
}

export interface ChildEventInvitationLabelQueryVariables {
  childId: string;
}
