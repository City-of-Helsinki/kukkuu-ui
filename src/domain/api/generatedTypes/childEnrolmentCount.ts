/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEnrolmentCount
// ====================================================

export interface ChildEnrolmentCount_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many times a single user can participate events per year. Changing this will not affect any existing enrolments.
   */
  enrolmentLimit: number;
}

export interface ChildEnrolmentCount_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many past enrolments child has this year.
   */
  pastEnrolmentCount: number | null;
  project: ChildEnrolmentCount_child_project;
}

export interface ChildEnrolmentCount {
  child: ChildEnrolmentCount_child | null;
}

export interface ChildEnrolmentCountVariables {
  childId: string;
}
