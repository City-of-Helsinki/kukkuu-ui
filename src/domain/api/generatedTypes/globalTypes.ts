/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

/**
 * An enumeration.
 */
export enum Language {
  EN = "EN",
  FI = "FI",
  SV = "SV",
}

export enum RelationshipTypeEnum {
  ADVOCATE = "ADVOCATE",
  OTHER_GUARDIAN = "OTHER_GUARDIAN",
  OTHER_RELATION = "OTHER_RELATION",
  PARENT = "PARENT",
}

export interface ChildInput {
  firstName?: string | null;
  lastName?: string | null;
  birthdate: any;
  postalCode?: string | null;
  relationship?: RelationshipInput | null;
}

export interface RelationshipInput {
  type?: RelationshipTypeEnum | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
