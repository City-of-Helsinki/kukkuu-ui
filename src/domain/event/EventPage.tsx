import { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { IconArrowLeft } from 'hds-react';

import styles from './event.module.scss';
import PageWrapper from '../app/layout/PageWrapper';
import {
  EventQuery,
  ExternalTicketSystemEventQuery,
} from '../api/generatedTypes/graphql';
import { OccurrenceEvent } from './types/OccurrenceQueryTypes';

type EventProps = {
  event:
    | EventQuery['event']
    | OccurrenceEvent
    | ExternalTicketSystemEventQuery['event'];
  children?: ReactElement | Array<ReactElement | false>;
  success?: ReactElement;
  backTo?: string;
};

const EventPage = ({ event, children, success, backTo }: EventProps) => {
  const { t } = useTranslation();
  if (!event) return <></>;

  const backgroundImageStyle = event.image
    ? {
        backgroundImage: `url("${event.image}")`,
      }
    : {
        backgroundColor: 'var(--color-summer)',
      };

  return (
    <>
      <div
        className={styles.heroWrapper}
        style={backgroundImageStyle}
        title={event.imageAltText || ''}
      >
        {success}
        <div className={styles.backButtonWrapper}>
          <div className={styles.backButtonInnerWrapper}>
            {backTo && (
              <Link
                aria-label={t('common.backButton.label')}
                className={styles.backButton}
                to={backTo}
              >
                <IconArrowLeft className={styles.backButtonIcon} />
              </Link>
            )}
          </div>
        </div>
      </div>

      <PageWrapper className={styles.wrapper} title={event.name || ''}>
        <div className={styles.eventWrapper} role="main">
          <div className={styles.event}>
            <div className={styles.heading}>
              <h1>{event.name}</h1>
            </div>
            {children}
          </div>
        </div>
      </PageWrapper>
    </>
  );
};

export default EventPage;
