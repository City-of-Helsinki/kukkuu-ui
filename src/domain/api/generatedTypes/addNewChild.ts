/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: addNewChild
// ====================================================

export interface addNewChild_addChild_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface addNewChild_addChild_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: addNewChild_addChild_child_project;
}

export interface addNewChild_addChild {
  child: addNewChild_addChild_child | null;
}

export interface addNewChild {
  /**
   * This mutation cannot be used before one has started using the service with "SubmitChildrenAndGuardianMutation".
   */
  addChild: addNewChild_addChild | null;
}

export interface addNewChildVariables {
  input: AddChildMutationInput;
}
