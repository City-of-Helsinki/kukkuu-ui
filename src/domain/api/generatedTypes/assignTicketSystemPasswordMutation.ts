/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssignTicketSystemPasswordMutationInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: assignTicketSystemPasswordMutation
// ====================================================

export interface assignTicketSystemPasswordMutation_assignTicketSystemPassword {
  /**
   * The assigned ticket system password
   */
  password: string | null;
}

export interface assignTicketSystemPasswordMutation {
  assignTicketSystemPassword: assignTicketSystemPasswordMutation_assignTicketSystemPassword | null;
}

export interface assignTicketSystemPasswordMutationVariables {
  input: AssignTicketSystemPasswordMutationInput;
}
