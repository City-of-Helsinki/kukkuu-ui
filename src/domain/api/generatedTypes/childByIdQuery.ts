/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { EventParticipantsPerInvite, RelationshipTypeEnum } from "./globalTypes";

// ====================================================
// GraphQL query operation: childByIdQuery
// ====================================================

export interface childByIdQuery_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface childByIdQuery_child_occurrences_edges_node_venue {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  address: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_event {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  /**
   * In minutes
   */
  duration: number | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * An unique encoded reference id
   */
  referenceId: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node_enrolments_edges_node | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges_node_enrolments_edges | null)[];
}

export interface childByIdQuery_child_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
  venue: childByIdQuery_child_occurrences_edges_node_venue;
  event: childByIdQuery_child_occurrences_edges_node_event;
  enrolments: childByIdQuery_child_occurrences_edges_node_enrolments;
}

export interface childByIdQuery_child_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node | null;
}

export interface childByIdQuery_child_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges | null)[];
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
}

export type childByIdQuery_child_availableEventsAndEventGroups_edges_node = childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode | childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode;

export interface childByIdQuery_child_availableEventsAndEventGroups_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_availableEventsAndEventGroups_edges_node | null;
}

export interface childByIdQuery_child_availableEventsAndEventGroups {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_availableEventsAndEventGroups_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node | null;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges_node_occurrences_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
  occurrences: childByIdQuery_child_pastEvents_edges_node_occurrences;
}

export interface childByIdQuery_child_pastEvents_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node | null;
}

export interface childByIdQuery_child_pastEvents {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges | null)[];
}

export interface childByIdQuery_child_relationships_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  type: RelationshipTypeEnum | null;
}

export interface childByIdQuery_child_relationships_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_relationships_edges_node | null;
}

export interface childByIdQuery_child_relationships {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_relationships_edges | null)[];
}

export interface childByIdQuery_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: childByIdQuery_child_project;
  occurrences: childByIdQuery_child_occurrences;
  availableEventsAndEventGroups: childByIdQuery_child_availableEventsAndEventGroups | null;
  pastEvents: childByIdQuery_child_pastEvents | null;
  relationships: childByIdQuery_child_relationships;
}

export interface childByIdQuery {
  /**
   * The ID of the object
   */
  child: childByIdQuery_child | null;
}

export interface childByIdQueryVariables {
  id: string;
}
