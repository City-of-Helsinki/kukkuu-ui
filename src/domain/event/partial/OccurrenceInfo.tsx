import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import joinClassNames from 'classnames';

import styles from './occurrenceInfo.module.scss';
import clockIcon from '../../../assets/icons/svg/clock.svg';
import calendarIcon from '../../../assets/icons/svg/calendar.svg';
import locationIcon from '../../../assets/icons/svg/location.svg';
import personIcon from '../../../assets/icons/svg/person.svg';
import { formatOccurrenceTime } from '../EventUtils';
import { childByIdQuery_child_occurrences_edges_node as OccurrenceType } from '../../api/generatedTypes/childByIdQuery';
import { occurrenceQuery_occurrence as OccurrenceQueryType } from '../../api/generatedTypes/occurrenceQuery';
import { formatTime, newMoment } from '../../../common/time/utils';
import { DEFAULT_DATE_FORMAT } from '../../../common/time/TimeConstants';
import InfoItem from './InfoItem';

type Props = {
  className?: string;
  occurrence: OccurrenceQueryType | OccurrenceType;
  show?: string[];
};

export type InfoItem = {
  id: string;
  className?: string;
  iconSrc: string;
  iconAlt?: string;
  label: string;
};

const OccurrenceInfo: FunctionComponent<Props> = ({
  className,
  occurrence,
  show = ['time', 'duration', 'participants', 'venue'],
}) => {
  const { t } = useTranslation();

  const infoItems: InfoItem[] = [
    {
      id: 'time',
      iconSrc: calendarIcon,
      iconAlt: t('event.register.occurrenceTableHeader.date'),
      label: formatTime(newMoment(occurrence.time), DEFAULT_DATE_FORMAT),
    },
    {
      id: 'duration',
      iconSrc: clockIcon,
      iconAlt: t('event.register.occurrenceTableHeader.time'),
      label: formatOccurrenceTime(occurrence.time, occurrence.event.duration),
    },
    {
      id: 'participants',
      iconSrc: personIcon,
      iconAlt: t('event.register.participants'),
      label: t(
        `event.participantsPerInviteEnum.${occurrence.event.participantsPerInvite}`
      ),
    },
    {
      id: 'venue',
      iconSrc: locationIcon,
      iconAlt: t('event.register.occurrenceTableHeader.venue'),
      label: occurrence.venue.name || '',
    },
  ];

  return (
    <div className={joinClassNames(className, styles.row)}>
      {infoItems.map(
        (item, index) =>
          show.some((id) => id === item.id) && (
            <InfoItem
              key={index}
              className={joinClassNames(
                className,
                styles.label,
                item.className
              )}
              iconSrc={item.iconSrc}
              iconAlt={item.iconAlt}
              label={item.label}
            />
          )
      )}
    </div>
  );
};

export default OccurrenceInfo;
