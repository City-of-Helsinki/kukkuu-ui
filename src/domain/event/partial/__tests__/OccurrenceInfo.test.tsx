import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import OccurrenceInfo from '../OccurrenceInfo';
import { mockedOccurrenceNode } from '../../__tests__/EventPage.test';
import { childByIdQuery_child_enrolments_edges_node_occurrence as CbiOccurrenceType } from '../../../api/generatedTypes/childByIdQuery';
import { EventParticipantsPerInvite } from '../../../api/generatedTypes/globalTypes';

const z: CbiOccurrenceType = {
  id: 'aa',
  time: '2020-03-08T04:00:00+00:00',
  event: {
    id: 'zzaaz',
    name: 'event name',
    image: 'a',
    imageAltText: 'b',
    shortDescription: 'd',
    duration: 12,
    participantsPerInvite: EventParticipantsPerInvite.FAMILY,
  },
  venue: {
    id: 'auppss',
    description: 'some venue description',
    name: 'Musiikkitalo',
    address: 'Urho Kekkosen katu 12',
  },
};

it('renders occurrence snapshot correctly', () => {
  const element = shallow(<OccurrenceInfo occurrence={mockedOccurrenceNode} />);
  expect(toJson(element)).toMatchSnapshot();
});

it('renders childByIdQuery occurrence snapshot correctly', () => {
  const element = shallow(<OccurrenceInfo occurrence={z} />);
  expect(toJson(element)).toMatchSnapshot();
});
