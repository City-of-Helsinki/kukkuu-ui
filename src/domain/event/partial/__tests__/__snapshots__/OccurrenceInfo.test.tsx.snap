// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders childByIdQuery occurrence snapshot correctly 1`] = `
<div
  className="row"
>
  <InfoItem
    className="label"
    iconAlt="Päivämäärä"
    iconSrc="calendar.svg"
    key="0"
    label="8.3.2020"
  />
  <InfoItem
    className="label"
    iconAlt="Kellonaika"
    iconSrc="clock.svg"
    key="1"
    label="06.00 - 06.12"
  />
  <InfoItem
    className="label"
    iconAlt="Osallistujat"
    iconSrc="person.svg"
    key="2"
    label="Perhe"
  />
  <InfoItem
    className="label"
    iconAlt="Tapahtumapaikka"
    iconSrc="location.svg"
    key="3"
    label="Musiikkitalo"
  />
</div>
`;

exports[`renders occurrence snapshot correctly 1`] = `
<div
  className="row"
>
  <InfoItem
    className="label"
    iconAlt="Päivämäärä"
    iconSrc="calendar.svg"
    key="0"
    label="8.3.2020"
  />
  <InfoItem
    className="label"
    iconAlt="Kellonaika"
    iconSrc="clock.svg"
    key="1"
    label="06.00 - 06.12"
  />
  <InfoItem
    className="label"
    iconAlt="Osallistujat"
    iconSrc="person.svg"
    key="2"
    label="Perhe"
  />
  <InfoItem
    className="label"
    iconAlt="Tapahtumapaikka"
    iconSrc="location.svg"
    key="3"
    label="Musiikkitalo"
  />
</div>
`;

exports[`renders snapshot correctly 1`] = `
<Fragment>
  <div
    className="heroWrapper"
    style={
      Object {
        "backgroundImage": "url(\\"a\\")",
      }
    }
    title="b"
  >
    <div
      className="backButtonWrapper"
    >
      <div
        className="backButtonInnerWrapper"
      >
        <Button
          aria-label="Palaa"
          className="backButton"
          onClick={[Function]}
          variant="secondary"
        >
          <Icon
            className="backButtonIcon"
            src="arrowLeft.svg"
          />
        </Button>
      </div>
    </div>
  </div>
  <PageWrapper
    className="wrapper"
    title="event name"
  >
    <div
      className="eventWrapper"
      role="main"
    >
      <div
        className="event"
      >
        <div
          className="heading"
        >
          <h1>
            event name
          </h1>
        </div>
      </div>
    </div>
  </PageWrapper>
</Fragment>
`;
